activiti-core-common/activiti-expression-language/src/main/java/org/activiti/core/el/CustomFunctionProvider.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2010-2020 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.activiti.core.el;
+
+public interface CustomFunctionProvider {
+
+    void addCustomFunctions(ActivitiElContext elContext) throws NoSuchMethodException;
```

Comment:
Why do we need to throw `NoSuchMethodException ` here?

Reply:
It's thrown by `class.getMethod()` which is used to provide the method. I can catch it inside the implementation though.

Reply:
Removed.